### PR TITLE
Updating Docs

### PR DESCRIPTION
**Pull request by kcajmagic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,133 @@
...
 
-1. Import the public GPG key (replace `0.0.1-65` with the release you want)
+Talaria can send events originating from the device as well as emit its own events.
+When this occurs, the event is sent to [Caduceus](https://github.com/xmidt-org/caduceus)
 
-```
-rpm --import https://github.com/Comcast/talaria/releases/download/0.0.1-65/RPM-GPG-KEY-comcast-xmidt
-```
+Talaria has three API endpoints to interact with the devices connected to itself.
+A [XMiDT](link here) client should not directly query against a talaria.
```
Comment (schmidtw) [inline README.md]: client = cloud client
link = xmidt.io ?

File: README.md [hunk 2]
```diff
@@ -1,40 +1,133 @@
...
+Instead, they should send a request through [scytale](link here).
```
Comment (schmidtw) [inline README.md]: link: https://github.com/xmidt-org/scytale

File: README.md [hunk 3]
```diff
@@ -1,40 +1,133 @@
...
+
+#### Device Statistics - `/device/{deviceID}/stat` endpoint
+This will return the statistics of the connected device,
+including information such as uptime and bytes sent.
+
+#### Get Devices - `/devices` endpoint
+Returns all the connected devices and their statistics.
```
Comment (schmidtw) [inline README.md]: Returns all the actively connected devices.

File: README.md [hunk 4]
```diff
@@ -1,40 +1,133 @@
...
+
+:warning: _Warning_: this is an expensive request. Use with caution.
+This is not recommended to be used in production.
+
+#### Send WRP to Device - `/device/send` endpoint
+Send a WRP message to the device.
+Talaria will accept either `msgpack` or `json`.
```
Comment (schmidtw) [inline README.md]:
```
Talaria will accept a WRP message encoded in a valid WRP representation - generally `msgpack` or `json`.
```

File: README.md [hunk 5]
```diff
@@ -1,40 +1,133 @@
...
+If the message is `json` encoded, talaria will encode the payload as `msgpack`.
+If the device returns a message it will be encoded as the `accept` header.
```
Comment (schmidtw) [inline README.md]: The HTTP accept header or the WRP.Accept value?

File: README.md [hunk 6]
```diff
@@ -1,40 +1,133 @@
...
 
-2. Install the rpm with yum (so it installs any/all dependencies for you)
+### Control Devices
+A secondary function of talaria is to control the connected devices.
+For more information refer to [Control Server Docs](docs/control_server.md).
 
+#### Gate Devices - `/device/gate` endpoint
+Allow or Deny device to connect to the talaria instance
+
+#### Gate Devices - `/device/drain` endpoint
```
Comment (schmidtw) [inline README.md]: Gate --> Drain

File: docs/control_server.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+## Device Gate
+Talaria can be set to disallow incoming websocket connections.
+When the gate is closed, all incoming websocket connection requests are rejected with a **503** status.
+Talaria always starts with the gate open, allowing new websocket connections.
+Already connected websockets are not affected by closing the gate.
+
+The RESTful endpoint that controls this is `/api/v2/device/gate`.
+Note that the port for the control server is not the same as the port for the websocket server.
+
+* `GET host:port/api/v2/device/gate` returns a JSON message indicating the status of the gate.
```
Comment (schmidtw) [inline docs/control_server.md]: One thought: It may be worth calling `port` `control_port` throughout?

File: docs/control_server.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+If no parameters are supplied, all devices are drained as fast as possible.
+Several parameters may be supplied to customize the drain:
+
+    + `count`: The maximum number of websocket connections to close.
+    If this value is larger than the number of connections at the start of the job,
+    the current count of connections is used instead.
+    + `percent`: The percentage of connections to close.
+    The computation of how many connections is done once when the job is started.
+    If both `count` and `percent` are supplied, `count` is ignored.
+    + `rate`: The number of connections per unit time (tick) to close.
```
Comment (schmidtw) [inline docs/control_server.md]: The `rate` and `tick` are a bit tricky.  A couple of simple examples will help a lot.  I have accidentally set the wrong rate a few times.

File: docs/control_server.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+is open and has been open since the given timestamp.  
+Similarly, `"open": false` indicates that the gate is closed.
+* `POST/PUT/PATCH host:control_port/api/v2/device/gate?open=<boolean>` raises or lowers the gate.
+The value of the open parameter may be anything that `golang` will parse as a boolean (see [ParseBool](https://godoc.org/strconv#ParseBool)).
+This endpoint is idempotent.
+Any attempt to open the gate when it is already open or close it when it is already closed results in a **200** status.
+If this endpoint did change the status of the gate, a **201** status is returned.
+
+### Metrics
+
```
Comment (kristinaspring) [inline docs/control_server.md]:
```suggestion
```

File: docs/control_server.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+The RESTful endpoint that controls the connection drain is `/api/v2/device/drain`.
+Note that the control_port for the control server is not the same as the port for the websocket server.
+
+* `GET host:control_port/api/v2/device/drain` returns a JSON message indicating whether
+a drain job is active and the progress of the active job if one is running.
+If a drain has previously completed, the information about that job will be
+available via this endpoint until a new drain job is started.
+
+* `POST/PUT/PATCH host:control_port/api/v2/device/drain` attempts to start a drain job.
+This endpoint returns the same JSON message as a `GET` when it starts a drain job,
```
Comment (kristinaspring) [inline docs/control_server.md]: I thought I already commented here but what does the get return? Information about the new job or information about the previous job?

File: README.md [hunk 11]
```diff
@@ -1,40 +1,133 @@
...
+
+#### Drain Devices - `/device/drain` endpoint
+Remove the connected devices to the talaria instance
+
+## Build
+
+### Source
+
+In order to build from the source, you need a working Go environment with
+version 1.11 or greater. Find more information on the [Go website](https://golang.org/doc/install).
```
[the remark this answers is not in the document]
Comment (kcajmagic) [reply README.md]: @kristinaspring I am pretty sure go mods work in 1.11

**Pull request merged**